services/delegation/src/modules/delegation/delegation.service.ts

```diff
@@ -590,4 +604,69 @@ export class DelegationService {
       .getRawOne();
     return parseInt(d.total, 10) || 0;
   }
+
+  private async undelegateFromChain(
+    nodeId: number,
+    type: string,
+    address: string,
+    amount: BigNumber,
+  ) {
+    let undelegationBlockNumber;
+    const targetChain =
+      type === NodeType.MAINNET ? NodeType.MAINNET : NodeType.TESTNET;
+    if (type === NodeType.TESTNET) {
+      undelegationBlockNumber = await this.testnetBlockchainService.undelegate(
+        address,
+        amount,
+      );
+    }
+
+    if (type === NodeType.MAINNET) {
+      undelegationBlockNumber = await this.mainnetBlockchainService.undelegate(
+        address,
+        amount,
+      );
+    }
```

Comment:
No need to call undelegate. It will be called by the cron

Reply:
Didn't we agree upon creating these on the condition of `if (currentDelegation.gt(totalNodeDelegation))` at https://github.com/Taraxa-project/taraxa-platform/blob/cc79dc42d27e15e6df40cf5b34c74649584788f0/services/delegation/src/modules/delegation/delegation.service.ts#L345 and then let the corn handle only the confirmations?